### DISCUSSION TITLE
Disable webengine's autogain control

### DISCUSSION THREAD

**Discussion by nrdxp** [qutebrowser/qutebrowser#6628] [Q&A] [answered]
We use Google Meet at work, and because of that, I noticed that chromium (and by extension, qt5-webengine) adjusts the volume of the microphone automatically to close to 0, making it hard for my colleuges to hear me. At first I believed this was some sort of system configuration issue, but I tried everything I could think (just alsa, pulseaudio, pipewire) but nothing worked.

After trying to research the issue, it turns out this seems to actually be a feature built into chromium itself, with no way to disable it by default. After finding a [google forum post](https://support.google.com/chrome/thread/7542181/chrome-is-auto-adjusting-the-microphone-level?hl=en) from someone experiencing the same issue, I found an [extension](https://github.com/joeywatts/disable-autogain-control-extension) for chrome that disables this behavior.

Obviously we cannot use this extension directly in qutebrowser, but I was wondering if there might be a way to disable this behavior somehow, so I don't have to referee my volume control during every meeting :laughing:

**Comment by The-Compiler** [comment, marked as answer]
Looking at the source of that extension, it mainly seems to run [this script](https://github.com/joeywatts/disable-autogain-control-extension/blob/master/disableAutogain.js). You should be able to add an appropriate [metadata block](https://sourceforge.net/p/greasemonkey/wiki/Metadata_Block/) and then use qutebrowser's Greasemonkey support.

I'll also move this to a discussion, since I don't think there's anything actionable for qutebrowser here.

**Reply by nrdxp** [reply]
Thanks for the quick response! I'll give that a try.

**Reply by nrdxp** [reply]
Unless the behavior is just not happening atm, it seems that just dumping the script unmodified in greasemonkey directory was enough to resolve the issue.

**Reply by appleseedexm** [reply]
Qutebrowser only recently started supporting chromium's autogain for me, dont know if something changed or why that is. Fortunately enough, this still works flawlessly!
